dor/entrypoints/api/catalog.py

```diff
@@ -0,0 +1,51 @@
+import uuid
+
+from dor.config import config
+from dor.service_layer import catalog_service
+import sqlalchemy
+from fastapi import APIRouter, Depends, status
+from fastapi.responses import JSONResponse
+
+from dor.adapters.catalog import SqlalchemyCatalog
+
+
+catalog_router = APIRouter(prefix="/catalog")
+
+ENGINE = sqlalchemy.create_engine(config.get_database_engine_url())
+
+
+def get_db_session():
+    with sqlalchemy.orm.Session(ENGINE) as session:
+        yield session
+
+
+@catalog_router.get("/bins/{identifier}/")
+def get_bin_summary(identifier: str, session = Depends(get_db_session)) -> JSONResponse:
+    try:
+        uuid_identifier = uuid.UUID(identifier)
+    except ValueError:
+        return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content="Identifier is not a valid UUID.")
+
+    catalog = SqlalchemyCatalog(session)
+    bin = catalog.get(uuid_identifier)
+    if bin:
+        summary = catalog_service.summarize(bin)
+        return JSONResponse(status_code=status.HTTP_200_OK, content=summary)
```

Comment:
I think, but am not entirely sure, that if we return dataclasses -- or at least Pydantic models -- directly, FastAPI will generate OpenAPI schemas for the return result. A nice perk but perhaps not a must-have. Anyways, I think something like this would work:

```python
if !bin:
    raise HttpException(status.HTTP_404_NOT_FOUND)
return catalog_service.summarize(bin) # with summarize creating a BinSummary
```

Reply:
Yeah, this worked when I hacked something together. Something to discuss:
```python
@catalog_router.get("/bins/{identifier}/")
def get_bin_summary(identifier: str, session = Depends(get_db_session)) -> BinSummary:
    try:
        uuid_identifier = uuid.UUID(identifier)
    except ValueError:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Identifier is not a valid UUID.")

    catalog = SqlalchemyCatalog(session)
    bin = catalog.get(uuid_identifier)
    if bin:
        summary = catalog_service.summarize(bin)
        return summary
    else:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Item not found")
```

Reply:
I regret `BinSummary` and would wish it forgotten :-) These responses feel more like the result of a `JBuilder` template, than building another class. What's the schema that you get with `BinSummary`?